src/components/lensFollow/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+
+import { useFollow, useUnFollow } from "../../lib/useFollow";
+
+interface FollowComponentProps {
+  profileId?: string;
+  isFollowed: boolean;
+}
+
+export function FollowComponent({
+  profileId,
+  isFollowed,
+}: FollowComponentProps) {
+  const { mutateAsync: followUser } = useFollow();
+  const { mutateAsync: unFollowUser } = useUnFollow();
+  const [buttonState, setButtonState] = useState(isFollowed);
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+  const handleFollowClick = async () => {
+    setIsButtonDisabled(true);
+    try {
+      await followUser(profileId!);
+      setButtonState(true);
+    } catch (error) {
+      console.log("follow:error");
+    }
+    setIsButtonDisabled(false);
+  };
+  const handleUnFollowClick = async () => {
+    setIsButtonDisabled(true);
+    try {
+      await unFollowUser(profileId!);
+      setButtonState(false);
+    } catch (error) {
+      console.log("unfollow:error");
+    }
+    setIsButtonDisabled(false);
+  };
+  if (buttonState === false) {
```

Comment:
```suggestion
 <button
        className="h-8 items-center justify-center border-2 border-solid border-black p-2 text-xs font-bold leading-[0.875rem] text-black hover:bg-black hover:text-white"
        disabled={isButtonDisabled}
        onClick={isFollowed? handleUnFollowClick : handleFollowClick}
      >
       isFollowed? UnFollow:  Follow
      </button>
```

Handle the follow & unfollow in parent component, so this component don't need to create and track any states